kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala

```diff
@@ -70,6 +71,17 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
 
   private lazy val port: Int = conf.get(FRONTEND_REST_BIND_PORT)
 
+  private lazy val isSecurityEnabled = {
+    val authTypes = conf.get(AUTHENTICATION_METHOD).map(AuthTypes.withName)
+    !authTypes.contains(NONE) && authTypes != Set(NOSASL)
```

Comment:
Just a trivial question, is it possible to reuse the variables defined in `KyuubiAuthenticationFactory`.
```
  private val authTypes = conf.get(AUTHENTICATION_METHOD).map(AuthTypes.withName)
  private val none = authTypes.contains(NONE)
  private val noSasl = authTypes == Set(NOSASL)
```

Reply:
Thanks for pointing out it, actually, I found bunches of duplicated code in THRIFT-HTTP and RESTful protocols, I'm fixing the tests and reducing the code duplications now.

Reply:
FYI: I refactored code to reuse `getValidPasswordAuthMethod` as you suggested

Reply:
I mis-interpreted the suggestion, you suggested `AuthTypes` but I used `AuthMethods`, which cause `IllegalArgumentException("No valid Password Auth detected")` when configuring `kyuubi.authentication=KERBEROS`, will fix it soon.